src/main/java/com/google/sps/data/TextAnalyser.java

```diff
@@ -0,0 +1,169 @@
+*/
+public final class TextAnalyser {
+  private final String message;
+
+  public TextAnalyser(String message) {
+     this.message = message.toLowerCase();
+  }
+
+  public float getSentimentScore() throws IOException {
+    try (LanguageServiceClient languageService = LanguageServiceClient.create()) {
+      Document doc =
+        Document.newBuilder().setContent(message).setType(Document.Type.PLAIN_TEXT).build();
+      Sentiment sentiment = languageService.analyzeSentiment(doc).getDocumentSentiment();
+        
+      return sentiment.getScore();
+    }
+  }
+
+  public String getMood() throws IOException {
+    float score = getSentimentScore();
+    
+    if(score == 1) {
+      return "very happy";
+    }
+ 
+    if(score == -1) {
+      return "so pessimistic";
+    }
+ 
+    int position = 0;
+    String[] moods = new String[]{"neutral", "calm", "relaxed", "serene", "contented",
+                                  "joyful", "happy", "delighted", "excited", "thrilled",
+                                  "tense", "nervous", "stressed", "upset", "sad", 
+                                  "depressed", "bored", "fatigued", "pessimisctic"};
+ 
+    score = score * 10;
+    position = (int) score;
+ 
+    if(position < 0) {
+      position = position * (-1) + 9;
+    }
+ 
+    return moods[position];
+  }
+
+  private ClassifyTextResponse classify() {
+    try (LanguageServiceClient language = LanguageServiceClient.create()) {
+      Document document = 
+            Document.newBuilder().setContent(message).setType(Type.PLAIN_TEXT).build();
+      ClassifyTextRequest request = ClassifyTextRequest.newBuilder().setDocument(document).build();
+        
+      return language.classifyText(request);
+    } catch(Exception e) {
+      return null;
+    }
+  }
+
+  // list of all the categories that text is about
+  // check https://cloud.google.com/natural-language/docs/categories
+  public List<String> getCategories() {
+    ClassifyTextResponse response = classify();
+
+    if(response == null) {
+      return Collections.emptyList();
+    }
+      
+    ArrayList<String> categories = new ArrayList<String>();
+
+    for (ClassificationCategory category : response.getCategoriesList()) {
+      String[] listCategories = category.getName().split("/");
+      for (int i = 0; i < listCategories.length; i++) {
+        categories.add(listCategories[i].toLowerCase());
+      }
+    }
+
+    return categories;
+  }
+
+  public Set<String> getEvents() {
+    Set<String> events = new HashSet<String>();
+    String[] allEvents = new String[]{"birthday", "wedding", "baby shower", "love",
+                                    "congratulation", "travel", "good morning",
+                                    "gratitude", "job", "promotion",
+                                    "new", "welcome", "good evening", "good night",
+                                    "holiday"};
+    String copy = new String(message);
```

Comment:
Strings are immutable in Java. It is thus not necessary to make a copy. Just say message.toLowerCase() and assign it to some variable and it will just work (tm).